pkg/controller/nodeconfigpod/tune.go

```diff
@@ -0,0 +1,181 @@
+	scyllaCpus := cpuset.NewCPUSet()
+	for _, p := range scyllaPods {
+		if p.GetLabels()[naming.ClusterNameLabel] != "" {
+			scs, ok := scyllaContainerStatus(p)
+			if !ok || scs.State.Running == nil {
+				return cpuset.CPUSet{}, fmt.Errorf("scylla container in '%s/%s' pod is not running, will retry in a bit", p.Namespace, p.Name)
+			}
+			if p.Status.QOSClass == corev1.PodQOSGuaranteed {
+				containerCpuSet, err := scyllaContainerCpuSet(ctx, criClient, p, naming.HostFilesystemDirName)
+				if err != nil {
+					return cpuset.CPUSet{}, fmt.Errorf("failed to get cpuset of '%s/%s' Pod Scylla container: %w", p.Namespace, p.Name, err)
+				}
+				klog.V(4).InfoS("Scylla container cpuset", "cpuset", containerCpuSet.String(), "Pod", klog.KObj(p), "containerID", scs.ContainerID)
+				scyllaCpus = scyllaCpus.Union(containerCpuSet)
+			}
+		}
+	}
+
+	return scyllaCpus, nil
+}
+
+func scyllaDataDirHostPaths(ctx context.Context, criClient cri.Client, scyllaPods []*corev1.Pod) ([]string, error) {
+	dataDirs := strset.New()
+
+	for _, pod := range scyllaPods {
+		cid, err := scyllaContainerID(pod)
+		if err != nil {
+			return nil, fmt.Errorf("get Scylla container ID: %w", err)
+		}
+
+		cs, err := criClient.Inspect(ctx, cid)
+		if err != nil && !errors.Is(err, cri.NotFoundErr) {
+			return nil, fmt.Errorf("failed to inspect container %q, %w", cid, err)
+		}
+
+		if cs != nil {
+			for _, mount := range cs.Status.GetMounts() {
+				if mount.ContainerPath != naming.DataDir {
+					continue
+				}
+				dataDirs.Add(mount.HostPath)
+			}
+		}
+	}
+
+	return dataDirs.List(), nil
+}
+
+func cpusetFromCRI(ctx context.Context, client cri.Client, cid string) (cpuset.CPUSet, error) {
+	cs, err := client.Inspect(ctx, cid)
+	if err != nil {
+		return cpuset.CPUSet{}, fmt.Errorf("failed to inspect container %q, %w", cid, err)
+	}
+
+	if cs.Info.RuntimeSpec == nil {
+		return cpuset.CPUSet{}, nil
+	}
+
+	containerCpuSet, err := cpuset.Parse(cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus)
+	if err != nil {
+		return cpuset.CPUSet{}, fmt.Errorf("failed to parse container %q cpuset %q, %w", cid, cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus, err)
+	}
+
+	return containerCpuSet, nil
+}
+
+func scyllaContainerCpuSet(ctx context.Context, criClient cri.Client, pod *corev1.Pod, hostFilesystem string) (cpuset.CPUSet, error) {
+	cid, err := scyllaContainerID(pod)
+	if err != nil {
+		return cpuset.CPUSet{}, fmt.Errorf("get Scylla container ID: %w", err)
+	}
+
+	containerCpuSet, err := cpusetFromCRI(ctx, criClient, cid)
+	if err != nil && !errors.Is(err, cri.NotFoundErr) {
+		return cpuset.CPUSet{}, fmt.Errorf("get cpuset from CRI: %w", err)
+	}
+
+	if !containerCpuSet.IsEmpty() {
+		return containerCpuSet, nil
+	}
+
+	// On AWS and Minikube runtime information is not be available in CRI.
+	// Figure out assigned CPUs by manually reading cgroup fs.
+	klog.Info("Falling back to manual cpuset discovery via cgroups")
```

Comment:
ideally we'd find it in just one place then have 2 branches but I yet have to look into options

Reply:
cgroup is always there, the problem is that it's in different location on every cloud provider. That's why I first try CRI, if it's missing i fallback to hardcoded paths.